### ISSUE TITLE
Nestable Blade @extends @yield @section?

### ISSUE THREAD

**Issue by JerseyMilker** [laravel/framework#8970]
It seems `@extends` with `@yield` & `@section` only work properly between a master template and a direct child of that master template.  Maybe this is intentional, I don't know.

How hard would it be to allow an `@included` view partial to `@extend` another view partial, with same `@yield` & `@section` functionality between these partials?  Obviously `@include` works everywhere, no matter how far down it's nested through view partials.  It would be sick to use the mentioned directives in view partials, not just between a master template and it's direct child.

EDIT: I've uploaded a fresh Laravel app with example routes/views, and failing test case: https://github.com/JerseyMilker/laravel/tree/blade-nest-extends-yield-section

**Comment by franzliedke** [comment]
That is a very cool idea...

**Comment by JerseyMilker** [comment]
I agree.  `@extends` with `@yield` & `@section` are powerful organizational directives for templates.  It makes me sad that they don't work anywhere within a template structure like `@include` and all the other directives.  Top of my Christmas list.

**Comment by simonhamp** [comment]
That's a bloody good idea!

**Comment by GrahamCampbell** [comment]
I'll let Taylor know about this...

**Comment by JerseyMilker** [comment]
Thank you for considering it.  I think it would lend itself to component development with Blade.

**Comment by rkgrep** [comment]
Check this package https://github.com/RobinRadic/blade-extensions
There is `@partial('path.to.view')` syntax which does the same thing

**Comment by franzliedke** [comment]
Any thoughts, @taylorotwell?

**Comment by JerseyMilker** [comment]
@thecrypticace pointed out to me that everything works when using `@overwrite` instead of `@endsection`.  I tested it out, and everything works great!  I looked at source and found this:

```php
protected function compileEndsection($expression)
{
    return "<?php \$__env->stopSection(); ?>";
}

protected function compileStop($expression)
{
    return "<?php \$__env->stopSection(); ?>";
}

protected function compileOverwrite($expression)
{
    return "<?php \$__env->stopSection(true); ?>";
}
```
That `true` arg triggers section overwrite.  I think 9 times out of 10, people want compiler overwrite when they trying to redefine a section content with the same name.  Say I want to use `@section('body')` in various partials.  This would allow reusing `@extends @yield @section` directives everywhere without confusion.  If someone really needs that compiler overwrite protection, you can see `@stop` is available for this.

TL;DR... The functionality is already there with `@overwrite`.  I think this is just confusion on default behaviour of `@endsection`, not understanding that it prevents a section name from being used more than once by compiler.  Should that be default behaviour?

@taylorotwell What do you think about adding that `true` arg to `@endsection`? Is that too breaking a change?

**Comment by JerseyMilker** [comment]
I just submitted PR #9242 to allow for this.  Maybe for 5.2?
